scripts/jira_tickets/create_compara_release_JIRA_tickets.pl

```diff
@@ -71,183 +49,52 @@ sub main {
-    $parameters->{existing_tickets} = {map {(($parameters->{division} // '') . '--' . $_->{fields}->{summary}) => $_->{key}} @{$existing_tickets->{issues}}};
-    $logger->info('Existing tickets: ' . Dumper($parameters->{existing_tickets}));
-
-    # -----------------------
-    # create new JIRA tickets
-    # -----------------------
-    for my $ticket ( @{$tickets} ) {
-        my $ticket_key = Bio::EnsEMBL::Compara::Utils::JIRA->create_ticket( $ticket->{'jira'}, $parameters, $logger );
-        if ($ticket->{'subtasks'}) {
-            foreach my $subtask (@{$ticket->{'subtasks'}}) {
-                $subtask->{'jira'}->{'fields'}->{'parent'} = { 'key'  => $ticket_key };
-                my $subtask_key = Bio::EnsEMBL::Compara::Utils::JIRA->create_ticket( $subtask->{'jira'}, $parameters, $logger );
-            }
-        }
-    }
-}
-
-=head2 set_parameters
-
-  Arg[1]      : String $relco - a Regulation team member name or JIRA username
-  Arg[2]      : Integer $release - the EnsEMBL release version
-  Arg[3]      : String $password - user's JIRA password
-  Arg[4]      : String $tickets_json - path to the json file that holds the input
-  Arg[5]      : String $config - path to the config file holding handover dates
-  Arg[6]      : String $division
-  Arg[7]      : Bio::EnsEMBL::Utils::Logger $logger - object used for logging
-  Description : Makes sure that the parameters provided through the command line
-                are valid and assigns default values to the ones which where not 
-                supplied
-  Return type : Listref
-  Exceptions  : none
-
-=cut
-
-sub set_parameters {
-    my ( $relco, $release, $password, $tickets_json, $config, $division, $logger ) = @_;
-
-    $relco = $ENV{'USER'} if !$relco;
-    $relco = Bio::EnsEMBL::Compara::Utils::JIRA->validate_user_name( $relco, $logger );
-
-    my $user = Bio::EnsEMBL::Compara::Utils::JIRA->validate_user_name( $ENV{'USER'}, $logger );
-
-    $release = Bio::EnsEMBL::ApiVersion->software_version() if !$release;
-
-    my %capitalized_divisions = (
-        'grch37'        => 'GRCh37',
-        'ensembl'       => 'Ensembl',
-        'vertebrates'   => 'Vertebrates',
-        'plants'        => 'Plants',
-        'metazoa'       => 'Metazoa',
+    # Get a new Utils::JIRA object to create the tickets for the given relco,
+    # division and release
+    my $jira_adaptor = new Bio::EnsEMBL::Compara::Utils::JIRA(
+        -RELCO    => $relco,
+        -DIVISION => $division,
+        -RELEASE  => $release
     );
-    $division ||= '';
-    if (!$division) {
-        warn "No division given, will load the division-agnostic 'relco' JSON file";
-
-    } elsif (!$capitalized_divisions{lc $division}) {
-        $logger->error("Division '$division' not recognized", 0, 0);
-    }
-    $division = $division ? $capitalized_divisions{lc $division} : 'relco';
+    # If no division is given, set it to 'relco'
+    $division = $division ? lc $jira_adaptor->{_division} : 'relco';
```

Comment:
Note: `lc $jira_adaptor->{_division}` is the same as `ls $division`

Reply:
It is going to be soon, but this was meant to return "vertebrates" if `$division` was "ensembl".

Reply:
Ah, good point